client/executor/wasmtime/src/state_holder.rs

```diff
@@ -15,63 +15,29 @@
 // along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
 
 use crate::host::{HostContext, HostState};
-use std::cell::RefCell;
-use std::rc::Rc;
 
-/// A common place to store a reference to the `HostState`.
-///
-/// This structure is passed into each host function handler and retained in the implementation of
-/// `WasmRuntime`. Whenever a call into a runtime method is initiated, the host state is populated
-/// with the state for that runtime method call.
+scoped_tls::scoped_thread_local!(static HOST_STATE: HostState);
```

Comment:
You could have used the environmental crate for this.

This could also be problematic in the future when we wasmtime use for sandboxing for example. 

In a perfect world, we would pass the pointer to the host state as parameter to the functions, but yeah not sure if that works.

Reply:
> In a perfect world, we would pass the pointer to the host state as parameter to the functions, but yeah not sure if that works.

This requires changes to wasmtime, again.

> You could have used the environmental crate for this.
Will do in a separate PR

Reply:
Yeah both of this stuff was more me writing down some considerations. Nothing that should be done now :)

Reply:
Turns out `environmental` also uses TLS under the hood, so I don't see much difference. At least with `scoped_thread_local!` you can check if the value has been set.